lib/Partitioner/Partitioner.cpp

```diff
@@ -36,14 +36,23 @@ static uint64_t updateUsedMem(const std::set<Storage *> &usedStorage,
 }
 
 Partitioner::Partitioner(Module *parent, const std::vector<DeviceInfo> &devices,
-                         bool saturateHost)
-    : module_(parent), deviceInfo_(devices), saturateHost_(saturateHost) {
+                         std::vector<Backend *> backends, bool saturateHost,
```

Comment:
Why do you pass `backend` by value? It seems like it should be `const std::vector<Backend *> &backends`.